GliaWidgets/CallVisualizer/VisitorCode/VisitorCodeView.swift

```diff
@@ -0,0 +1,172 @@
+                    return nil
+                }
+            }
+
+            init(
+                viewType: ViewType = .alert(closeButtonTap: .nop),
+                style: AlertStyle = .initial,
+                visitorCode: String = "11111",
+                isPoweredByShown: Bool = true
+
+            ) {
+                self.viewType = viewType
+                self.style = style
+                self.visitorCode = visitorCode
+                self.isPoweredByShown = isPoweredByShown
+            }
+        }
+
+        var props: Props = Props() {
+            didSet {
+                renderProps()
+            }
+        }
+
+        private let titleLabel = UILabel().make { label in
+            label.numberOfLines = 0
+            label.textAlignment = .center
+        }
+
+        private lazy var stackView = UIStackView.make(.vertical, spacing: 24)(
+            titleLabel,
+            visitorCodeStack,
+            poweredBy
+        )
+
+        private lazy var poweredBy: PoweredBy = {
+            PoweredBy(style: props.style.poweredBy)
+        }()
+
+        private lazy var closeButton = Button(
+            kind: .visitorCodeClose,
+            tap: { [weak self] in
+                self?.props.closeButtonTap?()
+            }
+        ).make { button in
+            button.accessibilityIdentifier = "visitorCode_alert_close_button"
+        }
+
+        private lazy var visitorCodeStack = UIStackView.make(.horizontal, spacing: 8)()
+        private let contentInsets = UIEdgeInsets(top: 24, left: 24, bottom: 21, right: 24)
+        private let cornerRadius: CGFloat = 30
+
+        override func setup() {
+            super.setup()
+            clipsToBounds = true
+            layer.masksToBounds = false
+            layer.cornerRadius = cornerRadius
+            renderProps()
+        }
+
+        override func defineLayout() {
+            super.defineLayout()
+            addSubview(stackView)
+            addSubview(closeButton)
+            visitorCodeStack.distribution = .fill
+            visitorCodeStack.translatesAutoresizingMaskIntoConstraints = false
+            NSLayoutConstraint.activate([
+                stackView.leadingAnchor.constraint(equalTo: leadingAnchor, constant: contentInsets.left),
+                stackView.trailingAnchor.constraint(equalTo: trailingAnchor, constant: -contentInsets.right),
+                stackView.bottomAnchor.constraint(equalTo: bottomAnchor, constant: -contentInsets.bottom),
+                stackView.topAnchor.constraint(equalTo: topAnchor, constant: contentInsets.top),
+                visitorCodeStack.heightAnchor.constraint(equalToConstant: 60),
+                closeButton.trailingAnchor.constraint(equalTo: trailingAnchor, constant: -30),
+                closeButton.bottomAnchor.constraint(equalTo: bottomAnchor, constant: -26)
+            ])
+        }
+
+        override func layoutSubviews() {
+            super.layoutSubviews()
+            setNeedsDisplay()
```

Comment:
will this method call `setNeedsDisplay` again? from documentation:
> Call this method on your application’s main thread when you want to adjust the layout of a view’s subviews.
so perhaps it is not needed to call `setNeedsDisplay` inside, let's verify will it affect anything if we drop `setNeedsDisplay`.

Reply:
You were right. It was not needed anymore.

Reply:
fixed [676c74c](https://github.com/salemove/ios-sdk-widgets/pull/448/commits/676c74ceb781899d4ca4fa8a8ab684c8a1837308)